packages/use-optionally-controlled-state/src/index.tsx

```diff
@@ -1,23 +1,31 @@
-import {useState} from 'react';
+import {useState, useCallback, useRef} from 'react';
 import useConstant from 'use-constant';
 
+export type OptionalState<Value> =
```

Comment:
I wouldn't export this, I think it's sufficient to have this as a private type. Mostly because I'm a bit unsure about the naming. I think it's not "optional state" but rather the "options" the hook takes.

Reply:
yea but OptionalStateOptions was a bit too weird haha